docs/pages/reference.mdx

```diff
@@ -35,3 +35,58 @@
   ```
   pnpm mud set-version --commit <hash> && pnpm install
   ```
+
+## Debugging
+
+### Onchain components
+
+#### Foundry Debugging
+
+The MUD onchain components typically use the [Foundry development stack](https://book.getfoundry.sh/), meaning any of Foundry's debugging tools are available.
+Use this prodecure to view the output from [`anvil`](https://book.getfoundry.sh/reference/anvil/):
+
+1. Add `console.log` to your systems, edit their Solidity files.
+
+   - Import console.log.
+
+     ```solidity
+     import { console } from "forge-std/console.sol";
+     ```
+
+   - Add `console.log` statements inside your Solidity functions.
+
+     ```solidity
+     console.log("newValue:", newValue);
+     ```
+
+1. Run `anvil` manually.
+
+   ```sh
+   anvil -b 1 --block-base-fee-per-gas 0 | tee anvil.log
+   ```
+
+   You do not have to use `tee anvil.log`, but `anvil` logs so much that it is often useful to be able to look for information in the log file after it has scrolled off the screen.
+
+1. Edit `packages/contracts/package.json` to change the `scripts.dev` definition.
+
+   ```
+   "dev": "pnpm mud dev-contracts --tsgenOutput ../client/src/mud --rpc http://127.0.0.1:8545",
+   ```
+
+   Use `127.0.0.1`, there is a weird bug that makes it a problem to use `localhost` is some circumstances.
```

Comment:
I'm not sure this is worth noting. I think anvil's host just defaults to 127.0.0.1 (and printed in the logs), so it makes sense that you'd use that URL to access it.

I believe you can get localhost working if you run anvil with `anvil --host localhost`

Reply:
It doesn't work, even after upgrading to the latest `anvil`.

```
qbzzt1@penguin:~$ anvil --version
anvil 0.1.0 (1044042 2023-08-10T13:04:26.276364846Z)
qbzzt1@penguin:~$ anvil -b 1 --base-fee 0 --host localhost
error: invalid value 'localhost' for '--host <IP_ADDR>': invalid IP address syntax

For more information, try '--help'.
qbzzt1@penguin:~$ 
```

While `anvil -b 1 --base-fee 0 --host 0.0.0.0` works, it still doesn't let us connect to `localhost`. I think it's some kind of confusion related to IPv6.